src/group_theory/group_action.lean

```diff
@@ -8,137 +8,158 @@ import data.set.finite group_theory.coset
+(one : ∀ b : β, (1 : α) • b = b)
+(mul : ∀ (x y : α) (b : β), (x * y) • b = x • y • b)
 
-def orbit (a : β) := set.range (λ x : α, f x a)
+namespace monoid_action
 
-@[simp] lemma mem_orbit_iff {f : α → β → β} [is_monoid_action f] {a b : β} :
-  b ∈ orbit f a ↔ ∃ x : α, f x a = b :=
+variables (α) [monoid α] [monoid_action α β]
+
+def orbit (b : β) := set.range (λ x : α, x • b)
+
+variable {α}
+
+@[simp] lemma mem_orbit_iff {b₁ b₂ : β} : b₂ ∈ orbit α b₁ ↔ ∃ x : α, x • b₁ = b₂ :=
 iff.rfl
 
-@[simp] lemma mem_orbit (a : β) (x : α) :
-  f x a ∈ orbit f a :=
+@[simp] lemma mem_orbit (b : β) (x : α) : x • b ∈ orbit α b :=
 ⟨x, rfl⟩
 
-@[simp] lemma mem_orbit_self (a : β) :
-  a ∈ orbit f a :=
-⟨1, show f 1 a = a, by simp [is_monoid_action.one f]⟩
+@[simp] lemma mem_orbit_self (b : β) : b ∈ orbit α b :=
+⟨1, by simp [monoid_action.one]⟩
 
-instance orbit_fintype (a : β) [fintype α] [decidable_eq β] :
-  fintype (orbit f a) := set.fintype_range _
+instance orbit_fintype (b : β) [fintype α] [decidable_eq β] :
+  fintype (orbit α b) := set.fintype_range _
 
-def stabilizer (a : β) : set α :=
-{x : α | f x a = a}
+variable (α)
 
-@[simp] lemma mem_stabilizer_iff {f : α → β → β} [is_monoid_action f] {a : β} {x : α} :
-  x ∈ stabilizer f a ↔ f x a = a :=
-iff.rfl
+def stabilizer (b : β) : set α :=
+{x : α | x • b = b}
+
+variable {α}
+
+@[simp] lemma mem_stabilizer_iff {b : β} {x : α} :
+  x ∈ stabilizer α b ↔ x • b = b := iff.rfl
 
-def fixed_points : set β := {a : β | ∀ x, x ∈ stabilizer f a}
+variables (α) (β)
 
-@[simp] lemma mem_fixed_points {f : α → β → β} [is_monoid_action f] {a : β} :
-  a ∈ fixed_points f ↔ ∀ x : α, f x a = a := iff.rfl
+def fixed_points : set β := {b : β | ∀ x, x ∈ stabilizer α b}
 
-lemma mem_fixed_points' {f : α → β → β} [is_monoid_action f] {a : β} : a ∈ fixed_points f ↔
-  (∀ b, b ∈ orbit f a → b = a) :=
+variables {α} (β)
+
+@[simp] lemma mem_fixed_points {b : β} :
+  b ∈ fixed_points α β ↔ ∀ x : α, x • b = b := iff.rfl
+
+lemma mem_fixed_points' {b : β} : b ∈ fixed_points α β ↔
+  (∀ b', b' ∈ orbit α b → b' = b) :=
 ⟨λ h b h₁, let ⟨x, hx⟩ := mem_orbit_iff.1 h₁ in hx ▸ h x,
-λ h b, mem_stabilizer_iff.2 (h _ (mem_orbit _ _ _))⟩
+λ h b, mem_stabilizer_iff.2 (h _ (mem_orbit _ _))⟩
 
-lemma comp_hom [group γ] (f : α → β → β) [is_monoid_action f] (g : γ → α) [is_monoid_hom g] :
-  is_monoid_action (f ∘ g) :=
-{ one := by simp [is_monoid_hom.map_one g, is_monoid_action.one f],
-  mul := by simp [is_monoid_hom.map_mul g, is_monoid_action.mul f] }
+def comp_hom [monoid γ] (g : γ → α) [is_monoid_hom g] :
+  monoid_action γ β :=
+{ smul := λ x b, (g x) • b,
+  one := by simp [is_monoid_hom.map_one g, monoid_action.one],
+  mul := by simp [is_monoid_hom.map_mul g, monoid_action.mul] }
 
-end is_monoid_action
+end monoid_action
 
-class is_group_action [group α] (f : α → β → β) extends is_monoid_action f : Prop
+class group_action (α : Type u) (β : Type v) [group α] extends monoid_action α β
```

Comment:
I'd be very happy to get rid of this class and call everything `action` or something instead.